concrete-npe/src/tools.rs

```diff
@@ -0,0 +1,37 @@
+use concrete_commons::dispersion::DispersionParameter;
+use concrete_commons::numeric::UnsignedInteger;
+use std::ops::Mul;
+
+/// Computes the number of bits affected by the noise with a dispersion
+/// describing a normal distribution
+pub fn estimate_number_of_noise_bits<T, D>(dispersion: D) -> usize
+where
+    D: DispersionParameter,
+    T: UnsignedInteger,
+{
+    // get the standard deviation
+    let std_dev: f64 = dispersion.get_modular_standard_dev::<T>();
+
+    // the constant used for the computation
+    let z: f64 = 4.;
+    let tmp = f64::log2(std_dev * z);
+    if tmp < 0. {
+        // means no bits are affected by the noise in the integer representation
+        // (discrete space)
+        0usize
+    } else {
+        tmp.ceil() as usize
+    }
+}
+
+// Square function tool
+// Arguments:
+// * `x` - input
+// Output
+// * x^2
```

Comment:
There should not be any Arguments or Output section. It is impossible to maintain.